### PR TITLE
Task yc/j209 add basic components for mobile part i

### PR DESCRIPTION
**Pull request by EugeniusGR**

### CODE REVIEW

File: src/common/enums/enums.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { ButtonType } from './ui/ui';
```
Comment (eugeneprib) [inline src/common/enums/enums.ts]: We use index files as entry, so let's use index.ts

File: src/components/common/button/button.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { Button as ReactButton, View } from 'react-native';
```
Comment (eugeneprib) [inline src/components/common/button/button.tsx]:
```suggestion
import { Button, View } from 'react-native';
```

File: src/components/common/button/button.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+type Props = {
+    label: string;
```
Comment (eugeneprib) [inline src/components/common/button/button.tsx]:
```suggestion
    title: string;
```

File: src/components/common/button/button.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    onClick?: (evt: React.MouseEvent<HTMLButtonElement>) => void;
+};
+
+const Button: React.FC<Props> = ({
+    label,
```
Comment (eugeneprib) [inline src/components/common/button/button.tsx]:
```suggestion
    title,
```

File: src/components/common/button/button.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    onClick,
+}) => {
+    return (
+        <View>
```
Comment (eugeneprib) [inline src/components/common/button/button.tsx]: View is redundant here

File: src/components/common/button/button.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+            <ReactButton title={label} onPress={onClick} color={'#f3427f'} />
```
Comment (eugeneprib) [inline src/components/common/button/button.tsx]: Color should be populated from predefined values and should be selected by incoming prop param

File: src/components/common/button/button.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        </View>
+    );
+};
+
+export { Button };
```
Comment (eugeneprib) [inline src/components/common/button/button.tsx]: Export default

File: src/components/common/button/button.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
```
Comment (eugeneprib) [inline src/components/common/button/button.tsx]: No need for the `common` folder. The `components` folder should be enough for this

File: src/components/common/common.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+export { Input } from './input/input';
+export { Button } from './button/button';
+export { Link } from './link/link';
```
Comment (eugeneprib) [inline src/components/common/common.ts]: Use index files in here

File: src/components/common/input/input.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+const Input: React.FC<Props> = ({
+    label,
+    placeholder = '',
+    isSecure = false,
+    hasMultipleRows = false,
+}) => {
+    return (
+        <View>
```
Comment (eugeneprib) [inline src/components/common/input/input.tsx]: View is redundant here

File: src/components/common/input/input.tsx [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+            <Text style={styles.text}>{label}</Text>
```
Comment (eugeneprib) [inline src/components/common/input/input.tsx]: No need for label

File: src/components/common/input/input.tsx [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+            <TextInput placeholder={placeholder} multiline={hasMultipleRows} secureTextEntry={isSecure} />
+        </View>
+    );
+};
+
+const styles = StyleSheet.create({
```
Comment (eugeneprib) [inline src/components/common/input/input.tsx]: Put styles in another file

File: src/components/common/input/input.tsx [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    text: {
+        color: '#f3427f',
+        fontSize: 16,
+    }
+});
+
+export { Input };
```
Comment (eugeneprib) [inline src/components/common/input/input.tsx]: Default export

File: src/components/common/link/link.tsx [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+            if (supported) {
+                Linking.openURL(link);
+            } else {
+                console.log("Don't know how to open URI: " + link);
+            }
+        });
+    };
+
+    return (
+        <TouchableOpacity onPress={handleClick}>{children}</TouchableOpacity>
```
Comment (eugeneprib) [inline src/components/common/link/link.tsx]: No need for the children support here. Title should be fine.

File: src/components/common/link/link.tsx [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+    children,
+    link,
+}) => {
+
+    const handleClick = () => {
+        Linking.canOpenURL(link).then(supported => {
+            if (supported) {
+                Linking.openURL(link);
+            } else {
+                console.log("Don't know how to open URI: " + link);
```
Comment (eugeneprib) [inline src/components/common/link/link.tsx]: No console.log allowed

File: src/components/common/link/link.tsx [hunk 16]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
Comment (eugeneprib) [inline src/components/common/link/link.tsx]: Same for this file, use index instead.

File: src/components/common/link/link.tsx [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+            }
+        });
+    };
+
+    return (
+        <TouchableOpacity onPress={handleClick}>{children}</TouchableOpacity>
+    );
+};
+
+export { Link };
```
Comment (eugeneprib) [inline src/components/common/link/link.tsx]: Default export

**Review by eugeneprib — CHANGES_REQUESTED** [review]
Let's use yarn.lock as it was defined before

File: src/components/common/link/link.tsx (hunk 14 above)
Comment (EugeniusGR) [reply src/components/common/link/link.tsx]: I thought it can be like a wrapper for buttons

File: src/App.tsx [hunk 18]
```diff
@@ -1,7 +1,7 @@
 import 'react-native-gesture-handler'
 import React from 'react'
 import { Provider } from 'react-redux'
-import { SafeAreaView, Text, View } from 'react-native'
+import { Alert, SafeAreaView, Text, View } from 'react-native'
```
Comment (eugeneprib) [inline src/App.tsx]:
```suggestion
import { SafeAreaView, Text, View } from 'react-native'
```

File: src/common/enums/enums.ts [hunk 19]
```diff
@@ -0,0 +1 @@
+export { ButtonType, ButtonColor } from './ui/ui'
```
Comment (eugeneprib) [inline src/common/enums/enums.ts]: Should not be here

File: src/common/enums/ui/button-color.enum.ts [hunk 20]
```diff
@@ -0,0 +1,5 @@
+enum ButtonColor {
```
Comment (eugeneprib) [inline src/common/enums/ui/button-color.enum.ts]: We should not have a color and a type params separately, it should go as one, that defines the style for our custom button. Also it should be placed within the button component folder at least

File: src/components/button/button.tsx [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+import { ButtonColor } from '../../common/enums/enums'
+
+type Props = {
+  title: string
+  onClick?: (evt: React.MouseEvent<HTMLButtonElement>) => void
+  color?: ButtonColor
+}
+
+const CustomButton: React.FC<Props> = ({ title, onClick, color }) => {
+  return <Button title={title} onPress={onClick} color={color} />
```
Comment (eugeneprib) [inline src/components/button/button.tsx]: Button type (Primary, Secondary) should be here. We assume that it defines the styles of the button by the incoming type param

File: src/components/button/button.tsx [hunk 22]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import { Button } from 'react-native'
+import { ButtonColor } from '../../common/enums/enums'
+
+type Props = {
+  title: string
+  onClick?: (evt: React.MouseEvent<HTMLButtonElement>) => void
```
Comment (eugeneprib) [inline src/components/button/button.tsx]: Are you sure about the event type?

File: src/components/button/button.tsx [hunk 23]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
```
Comment (eugeneprib) [inline src/components/button/button.tsx]: Don't forget about making this file as index

**Review by eugeneprib — CHANGES_REQUESTED** [review]

File: src/components/button/button.tsx (hunk 22 above)
Comment (EugeniusGR) [reply src/components/button/button.tsx]: idk what exactly this button will do

**Comment by Denidzio** [comment]
image

File: src/components/input/Input.tsx [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+  label,
+  value,
+  isSecure = false,
+  hasMultipleRows = false,
+  style,
+  ...props
+}) => {
+  return (
+    <View style={style}>
+      <Text style={[styles.text]}>{label}</Text>
```
Comment (eugeneprib) [inline src/components/input/Input.tsx]: We don't need this label

**Review by eugeneprib — CHANGES_REQUESTED** [review]

**Pull request merged**